include error field in api chat error responses

## 🐞 Description

```/api/chat ``` returns error responses without a top-level error key.
When the request body is missing message, the API returns 400 with reply + type only. This breaks clients/tests that rely on a standard error field for failures.
---

## 🔁 Steps to Reproduce

Steps to reproduce the behavior:
1. Run the app (```python src/App.py```).
2. Send a POST request to ```/api/chat ``` with an empty JSON body ({}).
3. Check the JSON response.
4. Notice the response does not include an ```error``` field.

---

## 💻 Expected Behavior

For error cases (validation/server), API response should include a top-level error field, for example:

```{
  "error": "Please enter a message before sending.",
  "type": "validation_error"
}
```

---

## 🧠 Actual Behavior

API currently responds with:
```
{
  "reply": "Please enter a message before sending.",
  "type": "validation_error"
}
```
No error key is present.

---

## 🖼️ Screenshots / Logs (if applicable)

Pytest failure observed before fix:

```E AssertionError: assert 'error' in {'reply': 'Please enter a message before sending.', 'type': 'validation_error'}```


---

## 🧩 Environment

- OS: Windows 11  
- Browser: N/A (reproduced via API request)
- Node.js version: N/A (Python backend)
- Framework/Library version: Flask (Peacebot backend), pytest 9.0.3
---

## ✅ Checklist

- [✅] I have searched for existing issues before creating this one.  
- [✅] I have included all relevant details and reproduction steps.

--- a/src/App.py
+++ b/src/App.py
@@ -124,8 +124,10 @@ def api_chat():
         message = (data.get("message") or "").strip()
 
         if not message:
+            msg = "Please enter a message before sending."
             return jsonify({
-                "reply": "Please enter a message before sending.",
+                "error": msg,
+                "reply": msg,
                 "type": "validation_error"
             }), 400
 
@@ -148,8 +150,10 @@ def api_chat():
 
     except Exception as e:
         logger.exception(f"Error in api_chat: {e}")
+        msg = "Sorry, something went wrong on my side. Please try again later."
         return jsonify({
-            "reply": "Sorry, something went wrong on my side. Please try again later.",
+            "error": msg,
+            "reply": msg,
             "type": "server_error"
         }), 500
 

